src/resolvers/nodeResolver.ts

```diff
@@ -66,19 +81,29 @@ export class NodeResolver implements Resolver {
 		}
 	}
 
-	public needsEntry(filename: string): boolean {
+	public needsEntry(
+		entry: TarEntryPath,
+		specifiedDockerfilePath?: string,
+	): boolean {
+		// Note: tar files always use forward slash as path separators (regardless
+		// of platform/OS conventions), so the search for '/' instead of `path.sep`
+		// below is not a bug :-)
 		return (
-			filename === 'package.json' ||
-			filename === 'wscript' ||
-			_.endsWith(filename, '.gyp')
+			!specifiedDockerfilePath &&
+			(entry.path === 'package.json' ||
+				entry.path === 'wscript' ||
+				(!entry.path.includes('/') && entry.extension === '.gyp'))
```

Comment:
Two cases:
* If a specifiedDockeriflePath was specified, then this resolver (`NodeResolver`) will never match (`!specifiedDockerfilePath`). This was previously enforced through a filter over the `NodeResolver. allowSpecifiedDockerfile = false` member variable, which I have deleted for all resolvers.
* Otherwise, it will match `package.json`, `wscript` or `*.gyp` at the root of the project directory tree, as `entry.path` is the full path. This root-only rule was previously enforced in `resolveTarStreamOnEntry()` through a test: `if (path.normalize(name).split(path.sep).length === 1)`. That test had been added by a recent-ish pull request:
_fix: Correctly resolve contexts with multiple bundle descriptions_
https://github.com/balena-io-modules/resin-bundle-resolve/pull/32/files